### PR TITLE
Arch User Repository Instant Answer Spice

### PR DESCRIPTION
**Pull request by NateBrune**
# Spice Pull Request Template

Using this template will help us better understand your Instant Answer and assist you when necessary.  Simply copy and paste the markdown below into the description of your GitHub pull request and complete as appropriate.

**What does your Instant Answer do?**
Instant answer provides SPICE interface for Arch User Repository.

**What problem does your Instant Answer solve (Why is it better than organic links)?**
This instant answer makes it quick and easy to get the name of the correct package in the aur. I often find myself
searching the AUR for a library but I have to open the link to view the description to make sure it is infact 
the correct program/library

**What is the data source for your Instant Answer? (Provide a link if possible)**
aur.archlinux.org

**Why did you choose this data source?**
It is the most sensible, widley used, and stable copy of the AUR.

**Are there any other alternative (better) data sources?**
There is alternative mirrors, but they are often out of date and/or unreliable.

**What are some example queries that trigger this Instant Answer?**
"aur zsh-git" "Aur ssh"

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**
Arch Linux Users, and Programmers most likely

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**
No.

**Which existing Instant Answers will this one supersede/overlap with?**
None    

**Are you having any problems? Do you need our help with anything?**
My tile doesnt look as nice as some of the other Instant Answer tiles

**What are the terms of use for the API? Will DuckDuckGo need specific authorization (e.g. an API key)? Are there any costs associated with API usage?**
There are no costs assosiated with this Instant Answer. It is a completly open api.

**Where did you hear about DuckDuckHack? (For first time contributors)**
I just loved the idea of duckduckgo and 
saw that it has been getting a lot better and looked up if you had any programs and you did!

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**

The broken image link is a link to the reddit upvote icon. I think duckpan just doesn't like showing images so I'm gonna assume it works lol

## Checklist
Please place an 'X' where appropriate.

```
[X] Added metadata and attribution information
[] Wrote test file and added to t/ directory
[X] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
[X] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
[] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        [X] Google Chrome
        [X] Firefox
        [] Opera
        [] IE 10

    - Windows 7
        [] Google Chrome
        [] Firefox
        [] Opera
        [] IE 8
        [] IE 9
        [] IE 10

    - Windows XP
        [] IE 8

    - Mac OSX
        [] Google Chrome
        [] Firefox
        [] Opera
        [] Safari

    - iOS (iPhone)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - iOS (iPad)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - Android
        [] Firefox
        [] Native Browser
        [] Google Chrome
        [] Opera
        
    - Arch Linux
        [X] Firefox
```

### CODE REVIEW

File: lib/DDG/Spice/Aur.pm [hunk 1]
```diff
@@ -1,30 +1,41 @@
...
-attribution twitter => ['https://twitter.com/crazedpsyc', 'crazedpsyc'],
-            cpan => ['CRZEDPSYC', 'crazedpsyc'];
+attribution github => ["NateBrune", "Nate Brune"];
 
+# API endpoint - https://duck.co/duckduckhack/spice_attributes#spice-codetocode
 spice to => 'https://aur.archlinux.org/rpc.php?type=search&arg=$1&callback={{callback}}';
 
-triggers any => "aur", "archlinux package", "arch package", "arch linux package";
+# Triggers - https://duck.co/duckduckhack/spice_triggers
+triggers any => "Aur", "aur", "AUR";
```
Comment (puskin94) [inline lib/DDG/Spice/Aur.pm]: Triggers are case-insensitive! 
`triggers any => "aur";` will match `AuR`,`Aur`,`auR`,`aUr` & `AUR`

**Comment by NateBrune** [comment]
All fixed, my bad

**Comment by puskin94** [comment]
@NateBrune As I told, if you change `triggers any => "Aur";` into `triggers any => "aur";` everything will be good ;) Look at this:

aur

**Comment by moollaza** [comment]
@NateBrune, @puskin94 is correct. Just to clarify the triggers should always be lowercase, we internally lowercase the query when checking it for trigger matches.

File: lib/DDG/Spice/Aur.pm [hunk 2]
```diff
@@ -1,30 +1,41 @@
 package DDG::Spice::Aur;
+# ABSTRACT: Write an abstract here
+# Start at https://duck.co/duckduckhack/spice_overview if you are new
+# to instant answer development
 
-use strict;
```
Comment (moollaza) [inline lib/DDG/Spice/Aur.pm]: Please keep this in. Aside from the actual benefits of `use strict`, right now Github is having problems detecting the language we're using (Perl vs Perl6) so `use strict` ensures our code has proper Perl5 syntax highlighting. They currently don't highlight Perl6 code :disappointed:

File: lib/DDG/Spice/Aur.pm [hunk 3]
```diff
@@ -1,30 +1,41 @@
...
+description "Provides hook into Arch User Repository API";
+primary_example_queries "aur mopidy", "aur python2-twisted";
+# Uncomment and complete: https://duck.co/duckduckhack/metadata#category
+category "software";
+# Uncomment and complete: https://duck.co/duckduckhack/metadata#topics
+topics "computing";
 code_url "https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/lib/DDG/Spice/Aur.pm";
-topics "programming";
-category "programming";
-attribution twitter => ['https://twitter.com/crazedpsyc', 'crazedpsyc'],
```
Comment (moollaza) [inline lib/DDG/Spice/Aur.pm]: Please don't remove this attribution. We like to give credit where it's due for all our previous contributors as well.

File: lib/DDG/Spice/Aur.pm [hunk 4]
```diff
@@ -1,30 +1,41 @@
...
-            cpan => ['CRZEDPSYC', 'crazedpsyc'];
+attribution github => ["NateBrune", "Nate Brune"];
 
+# API endpoint - https://duck.co/duckduckhack/spice_attributes#spice-codetocode
 spice to => 'https://aur.archlinux.org/rpc.php?type=search&arg=$1&callback={{callback}}';
 
-triggers any => "aur", "archlinux package", "arch package", "arch linux package";
```
Comment (moollaza) [inline lib/DDG/Spice/Aur.pm]: I think we should keep the original trigger intact. It only helps trigger this IA more often, unless you think those lead to this over-triggering?

File: lib/DDG/Spice/Aur.pm [hunk 5]
```diff
@@ -1,30 +1,41 @@
...
 
+# Handle statement
 handle remainder => sub {
-    my $remainder = $_;
 
-    return unless $remainder;
+    # optional - regex guard
+    # return unless qr/^\w+/;
 
-    $remainder =~ s/^for\s//;
```
Comment (moollaza) [inline lib/DDG/Spice/Aur.pm]: It's probably useful to keep this line, or some similar logic that removes any weasel words so we're only sending the package name to the API.

File: share/spice/aur/aur.js [hunk 6]
```diff
@@ -1,22 +1,46 @@
...
+                    iconArrow: {
+                            url: DDG.get_asset_path('aur','arrow_up.png')
+                    }
+                };          
+            },
+  
+            templates: {
+                group: 'text',
+                options: {
+                    footer: Spice.aur.aur,
```
Comment (moollaza) [inline share/spice/aur/aur.js]: This is OK, but our naming convention is to name the file after the param. So `footer: Spice.aur.footer` would be ideal. Likewise the file should be renamed to `footer.handlebars`

File: share/spice/aur/aur.handlebars [hunk 7]
```diff
@@ -1,3 +1,3 @@
-{{Name}}-{{Version}}: {{Description}} {{#if License}}({{License}}){{/if}}
-{{#if OutOfDate}}(out of date){{/if}}
-{{#if URL}}[<a href="{{URL}}">official site</a>]{{/if}}
+<div>
+    <div><img src="{{iconArrow.url}}">  {{NumVotes}} {{#if Maintainer}} &middot  {{Maintainer}}{{/if}}</div>
```
Comment (moollaza) [inline share/spice/aur/aur.handlebars]: You can remove the `<div>`'s, we're already injecting this into a footer div

File: share/spice/aur/aur.js [hunk 8]
```diff
@@ -1,22 +1,46 @@
...
+                },
+                variants: {
+                tile: 'basic4'
```
Comment (moollaza) [inline share/spice/aur/aur.js]: please indent this :)

**Comment by NateBrune** [comment]
@moollaza Hey, thank you very much. I fixed everything that you wanted except the indentation because Im not really sure what you want indented. As it is right now, I think its properly indented.

**Comment by moollaza** [comment]
@NateBrune everything is looking much better, thanks!

Regarding the indentation, it's minor, but Line 39 should be indented 4 spaces:

```js
variants: {
tile: 'basic4'
},
```

Should be:

```js
variants: {
    tile: 'basic4'
},
```

**Comment by NateBrune** [comment]
@moollaza boom! pow! zap! 
All of the worlds problems are gone. What a wonderful commit.

File: t/Aur.t [hunk 9]
```diff
@@ -5,27 +5,22 @@ use warnings;
...
         caller => 'DDG::Spice::Aur'
     ),
-    'archlinux package powermate' => test_spice(
-        '/js/spice/aur/powermate',
+    'Arch Package ssh' => test_spice(
+        '/js/spice/aur/ssh',
         call_type => 'include',
         caller => 'DDG::Spice::Aur'
     ),
-    'arch package powermate' => test_spice(
```
Comment (moollaza) [inline t/Aur.t]: It looks like these test should be valid again? Let's bring them back :)

**Comment by moollaza** [comment]
@abeyang this one is ready for a design pass :)

File: lib/DDG/Spice/Aur.pm [hunk 10]
```diff
@@ -1,30 +1,39 @@
...
-name "AUR";
+# Caching - https://duck.co/duckduckhack/spice_advanced_backend#caching
+spice is_cached => 1; 
+
+# Metadata.  See https://duck.co/duckduckhack/metadata for help in filling out this section.
+name "Aur";
+source "aur.archlinux.org";
+icon_url "http://upload.wikimedia.org/wikipedia/commons/thumb/a/a5/Archlinux-icon-crystal-64.svg/1000px-Archlinux-icon-crystal-64.svg.png";
+description "Provides hook into Arch User Repository API";
+primary_example_queries "aur mopidy", "aur python2-twisted";
```
Comment (moollaza) [inline lib/DDG/Spice/Aur.pm]: Look like there's no API results for "aur python2-twisted" -- let's change that to a query that works :) 

We usually like to have passing tests for the example queries too.

**Comment by NateBrune** [comment]
High school Spanish skills on fleek. @moollaza  @abeyang 
That was a dumb mistake. python2-twisted is a regular arch package, not an AUR package.

**Comment by zekiel** [comment]
thanks for this! @NateBrune

**Comment by NateBrune** [comment]
My pleasure! @zekiel

**Comment by NateBrune** [comment]
Uh... I broke it but Im not really sure why or how :/ 
I just git pulled to catch up to the repo because it said I couldnt merge and then It broke. @moollaza

**Comment by moollaza** [comment]
@NateBrune it's no problem. You accidentally committed a file with merge conflicts. Take a look at the diff page above -- you'll see the .pm file has `HEAD>>>>>>>>>>` you need to remove those lines and keep what should be there.

If you click on the details and scroll way down you can see what failed and why. The other failures are due to line-ending bugs -- outside your control.

**Comment by abeyang** [comment]
Sorry for coming in so late -- looks good! :+1:

**Comment by NateBrune** [comment]
Yay! @moollaza is there anything else that _I_ need to do at this point?

**Comment by moollaza** [comment]
@NateBrune congrats! It's live: https://duckduckgo.com/?q=aur+mopidy&ia=software

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's & Issues or even submit more Instant Answers!

**Comment by moollaza** [comment]
Apologies for the delay!

**Comment by NateBrune** [comment]
Hi sorry for digging up this old pull request but I just realized you guys where hiring and realized I needed to mention @nilnilnil to help myself out. Thanks again, love you guys!

**Pull request merged**